troncos/contrib/asgi/logging/middleware.py

```diff
@@ -121,41 +121,23 @@ async def __call__(
         async def wrapped_send(message: dict[str, Any]) -> None:
             if "status" in message:
                 status[0] = message.get("status", 0)
-
             await send(message)
 
-            # "more_body" is used if there is still data to send. Log the request
-            # when "http.response.body" has no more data left to send in the response.
-            if message.get("type") == "http.response.body" and not message.get(
-                "more_body", False
-            ):
-                extra = {
-                    "http_client_addr": str(client_ip) if client_ip else "NO_IP",
-                    "http_method": method,
-                    "http_path": path,
-                    "http_version": http_version,
-                    "http_status_code": status[0],
-                    "duration": time.perf_counter() - start_time,
-                }
-                self._access.info(
-                    "",
-                    **extra,
-                )
```

Comment:
For some reason, when running FastAPI, the structlog trace injection processor could not get the trace context inside this `wrapped_send` method. 

This `wrapped_send` method is only really getting the http status code. So I moved the logging out of the method to the bottom of the middleware call, and everything works fine.

I also think this is cleaner!

Reply:
Hmm, this seems brittle.

When I first started fixing velodrome, the trace context was only accessible outside `wrapped_send`, now I did some refactoring (seemingly unrelated to this middleware) and the opposite is true, the trace context is only accessible inside `wrapped_send`.

If the route throws an error everything is fine. Any insights you have on this?

Reply:
The span (in some cases) is finished when calling `await send(message)`, causing the trace id not to be logged: https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/contrib/asgi/middleware.py#L220-L233

Reply:
Only way I could make this reliable is to explicitly log the trace id in the middleware. WDYT?

Reply:
Looks a bit weird, but I guess it works. I think you should add a comment describing why you use current_trace_context directly here. It's easy to refactor this away later if we don't mention why we have this directly here.